src/algebra/monoid_algebra.lean

```diff
@@ -667,6 +718,39 @@ f.single_mul_apply_aux r _ _ _ $ λ a, by rw [zero_add]
 
 end misc_theorems
 
+end add_monoid_algebra
+
+/-!
+#### Conversions between `add_monoid_algebra` and `monoid_algebra`
+
+While we were not able to define `add_monoid_algebra k G = monoid_algebra k (multiplicative G)` due
+to definitional inconveniences, we can still show the types are isomorphic.
+-/
+
+/-- The equivalence between `add_monoid_algebra` and `monoid_algebra` in terms of `multiplicative` -/
+protected def add_monoid_algebra.to_multiplicative [semiring k] [add_monoid G] :
+  add_monoid_algebra k G ≃+* monoid_algebra k (multiplicative G) :=
+{ to_fun := (finsupp.dom_congr multiplicative.of_add),
```

Comment:
@jcommelin, can you explain this line to me? It seems odd to repeat this both in the `..` below and in `to_fun` here.

Reply:
The benefit is that you don't get `blabla.to_fun` in the goals of the remain fields. But maybe in this case it doesn't even matter. (Suppose that you want to call `simp` in the proof of `map_mul'`, then it would be useful to have a "clean" goal.)

Reply:
Feel free to remove the line if it doesn't break the proof.

Reply:
It doesn't break the proof :)